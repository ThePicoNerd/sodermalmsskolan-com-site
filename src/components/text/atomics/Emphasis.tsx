import { Theme } from "@emotion/react";
import React, { FunctionComponent } from "react";
import { HTMLElementProps } from "../../../styles/overrides";

/**
 * Some bit of text that is REALLY important.
 */
const Emphasis: FunctionComponent<HTMLElementProps<HTMLElement>> = (
  props,
) => (
  <em
    css={(theme: Theme) => ({
      color: theme.color.text.primary,
      fontStyle: "italic",
    })}
    {...props}
  />
);

export default Emphasis;
