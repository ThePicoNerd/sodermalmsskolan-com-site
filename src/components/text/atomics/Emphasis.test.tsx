import { render } from "@testing-library/react";
import React from "react";
import Emphasis from "./Emphasis";

describe("emphasis", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Emphasis>Important text</Emphasis>);

    expect(asFragment()).toMatchSnapshot();
  });
});
