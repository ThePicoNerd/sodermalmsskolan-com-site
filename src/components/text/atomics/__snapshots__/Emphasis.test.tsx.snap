// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emphasis renders correctly 1`] = `
.emotion-0 {
  color: var(--color-text-primary);
  font-style: italic;
}

<em
  className="emotion-0"
>
  Important text
</em>
`;
